spyder/plugins/editor/plugin.py

```diff
@@ -1144,23 +1228,61 @@ def get_plugin_actions(self):
             show_codestyle_warnings_action,
             show_docstring_warnings_action,
             underline_errors,
-            MENU_SEPARATOR,
+        ]
+        source_menu_linting_actions = [
             self.todo_list_action,
             self.warning_list_action,
             self.previous_warning_action,
             self.next_warning_action,
-            MENU_SEPARATOR,
+        ]
+        source_menu_cursor_actions = [
             self.previous_edit_cursor_action,
             self.previous_cursor_action,
             self.next_cursor_action,
-            MENU_SEPARATOR,
+        ]
+        source_menu_formatting_actions = [
             eol_menu,
             trailingspaces_action,
             fixindentation_action,
             self.formatting_action
         ]
-        self.main.source_menu_actions = (
-            source_menu_actions + self.main.source_menu_actions)
+        if mainmenu:
+            source_menu = mainmenu.get_application_menu(
+                ApplicationMenus.Source
+            )
+            source_menu.aboutToShow.connect(self.refresh_formatter_name)
+            for option_item in source_menu_option_actions:
+                mainmenu.add_item_to_application_menu(
+                    option_item,
+                    omit_id=True,
+                    menu_id=ApplicationMenus.Source,
+                    section=SourceMenuSections.Options,
+                    before_section=SourceMenuSections.CodeAnalysis
```

Comment:
```suggestion
                    before_section=SourceMenuSections.Linting
```

Be more specific about the section before which this one needs to be placed on.